### PR TITLE
sublime-music: 0.11.7 -> 0.11.10

### PR DESCRIPTION
**Pull request by sumnerevans**
###### Motivation for this change

This PR

* Updates Sublime Music to v0.11.10
* Adds functionality to install the `.desktop` and `.metainfo.xml` files as well as the icon for Sublime Music. I had to change the source to use the `.tar.gz` from the Releases page instead of the source from PyPi because the tar file has those additional resources.
* Adds me (@sumnerevans) as a maintainer for the `sublime-music` package. I'm the creator of Sublime Music, and I recently switched to NixOS, so I figured it would be good to add myself to the maintainers list.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sumnerevans** [comment]
Just added a commit to bump to the latest release of Sublime Music: https://gitlab.com/sublime-music/sublime-music/-/releases/v0.11.10

**Comment by sumnerevans** [comment]
@albakham when you have a chance.

**Comment by albakham** [comment]
@jtojnar

File: pkgs/applications/audio/sublime-music/default.nix [hunk 1]
```diff
@@ -9,11 +9,11 @@
 
 python3Packages.buildPythonApplication rec {
   pname = "sublime-music";
-  version = "0.11.7";
+  version = "0.11.10";
 
-  src = python3Packages.fetchPypi {
-    inherit pname version;
-    sha256 = "1x6b02gw46gp6qcgv67j7k3gr1dpfczbyma6dxanag8pnpqrj8qi";
+  src = fetchurl {
```
Comment (jtojnar) [inline pkgs/applications/audio/sublime-music/default.nix:14]: You can use `fetchFromGitLab`.

File: pkgs/applications/audio/sublime-music/default.nix [hunk 2]
```diff
@@ -51,10 +52,20 @@ python3Packages.buildPythonApplication rec {
   # https://github.com/NixOS/nixpkgs/issues/56943
   strictDeps = false;
 
+  postInstall = ''
+    install -Dm444 sublime-music.desktop      -t $out/share/applications
```
Comment (jtojnar) [inline pkgs/applications/audio/sublime-music/default.nix:56]: Would be nice if upstream install script could install these but does not look like poetry supports that at the moment: https://github.com/python-poetry/poetry/issues/890

File: pkgs/applications/audio/sublime-music/default.nix (hunk 1 above)
Comment (sumnerevans) [reply pkgs/applications/audio/sublime-music/default.nix]: Good callout. Changed to use that.

File: pkgs/applications/audio/sublime-music/default.nix (hunk 2 above)
Comment (sumnerevans) [reply pkgs/applications/audio/sublime-music/default.nix:58]: I don't want to hard-code the installation of the `.desktop` and icon files into the upstream package. Even if Poetry supported it, I think it would cause issues if users do a `--user` install.

Comment (jtojnar) [reply pkgs/applications/audio/sublime-music/default.nix:58]: They would be installed to `$prefix` so for user install, it would be under `~/.local`.

Comment (sumnerevans) [reply pkgs/applications/audio/sublime-music/default.nix:58]: Hmm... I haven't really seen any examples of a Python package actually respecting that. Happy to integrate something along those lines if you can point me to it.

Comment (jtojnar) [reply pkgs/applications/audio/sublime-music/default.nix:58]: I believe that if you make the paths relative like

https://github.com/MeanEYE/Sunflower/blob/3c45a4cbd28b785d3d49f3fc0dca5e8227ee5d29/setup.py#L29-L33

it will just install it to whatever path user specifies with `--prefix`. It is setuptools specific, though.

Most Python projects I have seen migrated to Meson. For example: https://gitlab.gnome.org/GNOME/gnome-music

Comment (jtojnar) [reply pkgs/applications/audio/sublime-music/default.nix:58]: Looks like quodlibet uses gdist and poetry though: https://github.com/quodlibet/quodlibet/blob/e6b2c65eada1bc9625327ab560029215f74fc67b/setup.py#L94-L121

Comment (sumnerevans) [reply pkgs/applications/audio/sublime-music/default.nix:58]: @jtojnar I opened this issue on the Sublime Music repo: https://gitlab.com/sublime-music/sublime-music/-/issues/282

I just migrated Sublime Music from setuptools+Pipenv to poetry, and I really don't want to have to make another major infrastructure change anytime soon. I can try to figure out how to make Poetry install the files in the correct locations, but that's fairly involved and is going to be low priority considering there are still many user-visible bugs and missing features in Sublime Music that need to be solved before I try and improve this process.

Considering that the approach I've taken in this PR works, I am inclined to ask to have this merged and revisit this process later once I find the time (or once somebody comes along and contributes a fix for this to the Sublime Music project :joy:).

Comment (jtojnar) [reply pkgs/applications/audio/sublime-music/default.nix:58]: Sure, it can be fixed later.

**Comment by jtojnar** [comment]
Looks like it does not build on x86_64-linux but it does on aarch64-linux according to CI.

**Comment by sumnerevans** [comment]
OK, it looks like I've fixed the build, now.

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -9087,6 +9087,12 @@
     githubId = 65870;
     name = "Сухарик";
   };
+  sumnerevans = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:9090]: Please put this into a separate commit with the message ``maintainers: add sumnerevans``.

File: pkgs/applications/audio/sublime-music/default.nix [hunk 4]
```diff
@@ -1,4 +1,5 @@
-{ lib, python3Packages, gobject-introspection, gtk3, pango, wrapGAppsHook
+{ fetchFromGitLab, lib, python3Packages, gobject-introspection, gtk3, pango, wrapGAppsHook
+
 , chromecastSupport ? false
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/sublime-music/default.nix:3]:
```suggestion
{ fetchFromGitLab, lib, python3Packages, gobject-introspection, gtk3, pango, wrapGAppsHook
, chromecastSupport ? false
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 3 above)
Comment (sumnerevans) [reply maintainers/maintainer-list.nix:9090]: I think I fixed the commit history.
